include/vcpkg/base/messages.h

```diff
@@ -1601,6 +1732,12 @@ namespace vcpkg
         "vcpkg has crashed. Please create an issue at https://github.com/microsoft/vcpkg containing a brief summary of "
         "what you were trying to do and the following information.");
     DECLARE_MESSAGE(VcpkgInvalidCommand, (msg::command_name), "", "invalid command: {command_name}");
+    DECLARE_MESSAGE(
+        VcpkgJsonNoCommentMsg,
+        (),
+        "",
+        "\nvcpkg does not support c-style comments, however most objects allow $-prefixed fields to be used as "
```

Comment:
```suggestion
        "vcpkg does not support c-style comments, however most objects allow $-prefixed fields to be used as "
```

Better not add a trailing `\n` in the message itself.

Reply:
I don't know how this build is passing :)

Reply:
I think you need to repeat this comment in all the places it got used rather than putting them as separate messages. I know the old code did that but it's not necessarily safe to append these together and result in a coherent message.

Reply:
Also if this survives "vcpkg" and "msg" probably shouldn't be in the name of the constant :)